plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/BaseSqlServerTypeMapping.java

```diff
@@ -13,30 +13,146 @@
+
+    // no DST in 1970, but has DST in later years (e.g. 2018)
+    private final ZoneId vilnius = ZoneId.of("Europe/Vilnius");
+    private final LocalDateTime timeGapInVilnius = LocalDateTime.of(2018, 3, 25, 3, 17, 17);
+    private final LocalDateTime timeDoubledInVilnius = LocalDateTime.of(2018, 10, 28, 3, 33, 33, 333_000_000);
+
+    // minutes offset change since 1970-01-01, no DST
+    private final ZoneId kathmandu = ZoneId.of("Asia/Kathmandu");
+    private final LocalDateTime timeGapInKathmandu = LocalDateTime.of(1986, 1, 1, 0, 13, 7);
+
+    @BeforeClass
+    public void setUp()
+    {
+        checkIsGap(jvmZone, timeGapInJvmZone1);
+        checkIsGap(jvmZone, timeGapInJvmZone2);
+        checkIsDoubled(jvmZone, timeDoubledInJvmZone);
+
+        checkIsGap(vilnius, timeGapInVilnius);
+        checkIsDoubled(vilnius, timeDoubledInVilnius);
+
+        checkIsGap(kathmandu, timeGapInKathmandu);
+    }
+
+    @Test
+    public void testBasicTypes()
+    {
+        SqlDataTypeTest.create()
+                .addRoundTrip("boolean", "true", BOOLEAN)
+                .addRoundTrip("boolean", "false", BOOLEAN)
+                .addRoundTrip("bigint", "123456789012", BIGINT)
+                .addRoundTrip("integer", "123456789", INTEGER)
+                .addRoundTrip("smallint", "32456", SMALLINT, "SMALLINT '32456'")
+                .addRoundTrip("tinyint", "5", TINYINT, "TINYINT '5'")
+                .addRoundTrip("real", "123.456", REAL, "REAL '123.456'")
+                .addRoundTrip("double", "123.456789", DOUBLE, "DOUBLE '123.456789'")
+                .execute(getQueryRunner(), trinoCreateAsSelect("test_basic_types"));
+    }
+
+    @Test
+    public void testReal()
+    {
+        SqlDataTypeTest.create()
+                .addRoundTrip("real", "NULL", REAL, "CAST(NULL AS real)")
+                .addRoundTrip("real", "3.14", REAL, "REAL '3.14'")
+                .addRoundTrip("real", "3.1415927", REAL, "REAL '3.1415927'")
+                .execute(getQueryRunner(), onRemoteCreateAndInsert("test_real"));
+
+        SqlDataTypeTest.create()
+                .addRoundTrip("real", "NULL", REAL, "CAST(NULL AS real)")
+                .addRoundTrip("real", "3.14", REAL, "REAL '3.14'")
+                .addRoundTrip("real", "3.1415927", REAL, "REAL '3.1415927'")
+                .execute(getQueryRunner(), trinoCreateAsSelect("test_real"));
+    }
+
+    @Test
+    public void testDouble()
+    {
+        SqlDataTypeTest.create()
+                .addRoundTrip("double precision", "NULL", DOUBLE, "CAST(NULL AS double)")
+                .addRoundTrip("double precision", "1.0E100", DOUBLE, "1.0E100")
+                .execute(getQueryRunner(), onRemoteCreateAndInsert("test_double"));
+
+        SqlDataTypeTest.create()
+                .addRoundTrip("double", "NULL", DOUBLE, "CAST(NULL AS double)")
+                .addRoundTrip("double", "1.0E100", DOUBLE, "1.0E100")
```

Comment:
Does SQL Server support +/- infinity and NaN? If yes, can you add cases for that? Here, in real (float) and decimals.

Reply:
SQL Server doesn't support +/- infinity and NaN, added a comment.